### DISCUSSION TITLE
Sweet theme

### DISCUSSION THREAD

**Discussion by Gigas002** [prasanthrangan/hyprdots#653] [Q&A]
Hello, and thank you for your amazing project!
I wanted to port my favorite KDE Plasma theme to hyprland, and looked at your [Synth-Wave](https://github.com/prasanthrangan/hyprdots-mod) for a reference, but since my knowledge related to theming is close to 0 I have some questions to ask.

First of all, here's all the `sweet`-theme related repos:

- [sweet (nova branch)](https://github.com/EliverLara/Sweet/tree/nova) -- gtk2/3/4, gnome-shell, sddm, kvantum, cursors, etc
- [sweet-kde](https://github.com/EliverLara/Sweet-kde) -- kde-specific stuff (content's **not** the same, as in kde directory in repo above)
- [firefox-sweet-theme](https://github.com/EliverLara/firefox-sweet-theme) -- sweet theme for firefox
- [sweet-vscode](https://github.com/EliverLara/sweet-vscode) -- vscode theme
- [sweet-vscode-icons](https://github.com/EliverLara/sweet-vscode-icons) -- icons for sweet vscode theme
- [sweet-folders](https://github.com/EliverLara/Sweet-folders) -- folders icons
- [candy-icons](https://github.com/EliverLara/candy-icons) -- icons

As far as I could understand basing on `Synth-Wave` repo, the

[sweet/gnome-shell](https://github.com/EliverLara/Sweet/tree/nova/gnome-shell)
[sweet/gtk-2.0](https://github.com/EliverLara/Sweet/tree/nova/gtk-2.0)
[sweet/gtk-3.0](https://github.com/EliverLara/Sweet/tree/nova/gtk-3.0)
[sweet/gtk-4.0](https://github.com/EliverLara/Sweet/tree/nova/gtk-4.0)

Should go into something like `$theme/Source/arcs/Gtk_$theme.tar.xz/$theme/`

[candy-icons](https://github.com/EliverLara/candy-icons) -> `$theme/Source/arcs/Icon_$icon_theme.tar.xz/$icon_theme/`

[sweet/kde/Kvantum/Sweet](https://github.com/EliverLara/Sweet/tree/nova/kde/Kvantum/Sweet) -> `$theme/Configs/.config/Kvantum/$theme`

And also. judging by [readme in sweet repo](https://github.com/prasanthrangan/hyprdots-mod/blob/e80d533d4c1354f6aad34a07ffa0e4dabc376559/Configs/.config/hypr/themes/Synth-Wave.conf#L1) and a [Synth-Wave hyprland config](https://github.com/prasanthrangan/hyprdots-mod/blob/e80d533d4c1354f6aad34a07ffa0e4dabc376559/Configs/.config/hypr/themes/Synth-Wave.conf#L1), I should put these lines in `$theme/Configs/.config/hypr/themes/$theme.conf`:

```conf
exec = gsettings set org.gnome.desktop.interface icon-theme '$icon_theme'
exec = gsettings set org.gnome.desktop.interface gtk-theme '$theme'
exec = gsettings set org.gnome.desktop.wm.preferences theme '$theme'
```

That's about as much I could figure out. Now's the time for the questions:

1. Is this enough for a `themepatcher.sh` to grab a theme and it will work?
2. Can any of the other configs in sweet repos be somehow added/ported into theme? I especially worry about `qt5ct` stuff, since it's not existent into `sweet` project, so if I could look for colors in some of other existing config and like copy-paste it into `qt5ct` one I'd do so, but I just lack knowledge of what all these theme files do

**Comment by kRHYME7** [comment]
Wow! A new theme is coming up!

The patcher will find these files with that file system in the directory
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/eabf8430-0f9b-4cac-8e76-cb9bbbe7e9df)
This one too
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/8a305622-f7cb-481c-98c9-50e88ed2e318)


> Is this enough for a themepatcher.sh to grab a theme and it will work?

As long as you give a consistent Theme name I guess  it would work

> Can any of the other configs in sweet repos be somehow added/ported into theme? I especially worry about qt5ct stuff, since it's not existent into sweet project, so if I could look for colors in some of other existing config and like copy-paste it into qt5ct one I'd do so, but I just lack knowledge of what all these theme files do

As long as those colors are written inside ``` ".config/qt5ct/colors/$Fav_Theme.conf" ``` 

Please let us know if it is ready to test! Thanks. Also for reference visit @T-Crypt  repo as he had some Themes. 




Also if you have time, here's the On the making of the script 
https://github.com/prasanthrangan/hyprdots/discussions/468
https://github.com/prasanthrangan/hyprdots/discussions/526

**Reply by Gigas002** [reply]
Hello! Thank you for your answer. I've started working on a theme in [my repo](https://github.com/Gigas002/hyprdots-sweet) and tried it out. For now *some* parts are working, some are not, and some are not quite as good as they should be. I hope to figure it out. Would be glad, if users, that are more experienced in theming would help me!

**Reply by kRHYME7** [reply]
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/d3be3bf2-21d8-4280-aab2-633c2864069c)


``` Hyprdots theme patch "Sweet" "https://github.com/gigas002/hyprdots-sweet" "eliverLara.sweet-vscode~Sweet vscode" ```
For the extension you should add a default label into it. 

Fot this as Example:   "eliverLara.sweet-vscode~Sweet vscode"
eliverLara.sweet-vscode <<< This is the folder name also an the extension ID , I prefer it to be in lowercase incase as ~/.vscode/extensions directory list extension on this format

~   <<<  This is the delimitter to differentiate the ext ID and the actual Theme name. 

Sweet vscode <<< Upon deciding the extension theme the maintainer should explicitly give the default theme

also I recommen enclosing the extension with '  '   than "  " just in case. 

you can read its labels inside (for this extension)
~/.vscode/extensions/eliverlara.sweet-vscode-1.1.1/package.json 
You might see more or just a single theme.  The One we needed is the Label part.
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/aba3de53-9aad-4952-b3c8-9f008d78c1f9)


so here's the working one  

``` Hyprdots theme patch "Sweet" "https://github.com/gigas002/hyprdots-sweet" "eliverLara.sweet-vscode~Sweet vscode" ```

if you see that it still not working delete first the line of the theme in ``` ~/.config/hypr/theme.ctl ``` also maybe uninstall the extension...

OR user can just run ``` Hyprdots code select ``` and select the theme of the extension

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/945a5ecf-9766-403d-827b-00d9a3744eab)

Fore Demo: using Gruvbox
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/ab2be805-f0e8-4ef0-9ee4-a3d3db61381b)

**Reply by Gigas002** [reply]
I see, thank you for advices! Going to update the theme when I'll have some more time for testing

**Comment by Gigas002** [comment]
So, after a brief testing I have more questions, than answers. Sorry if these are stupid, but I couldn't figure it out by myself. Hope I'll get through it

1. None of the `gtk-4.0` themes doesn't work (tested with [gnome-calculator](https://archlinux.org/packages/extra/x86_64/gnome-calculator/)). Tried changing the symlink in `~/.config/gtk-4.0` to a different themes, but nothing helped. The symlink also wasn't updated, when I changed the theme to `sweet`
2. How can I configure the colors of `qt5ct`? Would be glad, if someone will show me the doc pages or something, because it's colors config contains some random repeating colors over and over without any notes for what it stands for. The `qt5ct`/`qt6ct` repo doesn't have such manual either...
3. How's the `waybar` theme changing works? I see that there's a `~/.config/waybar/themes/theme.css` file, which symlinks to `~/.config/waybar/themes/Wall-Dcol.css`, but the values inside this file are not the same, as in corresponding theme (e.g. when choosing `Synth-Wave` the colors in `Synth-Wave.css` and `Wall-Dcol.css` doesn't match)
4. How can I apply the cursor theme? It works for other themes, but not for mine. I tried setting `exec = gsettings set org.gnome.desktop.interface cursor-theme 'Sweet-cursors'` in `hypr/themes/Sweet.conf` but no success
5. `firefox` theme isn't applied on theme change. Though I doubt it's a theme issue, probably it's just not implemented yet
6. Can the `splashscreen` be customized with theme?

**Reply by kRHYME7** [reply]
@Gigas002  Hello! Sorry for the late reply. Github notifs are surely problematic. 
@prasanthrangan  I'll be pinging the Owner to this discussion maybe he can give more insights. I'll try to answer some of this questions too .

**Reply by Gigas002** [reply]
Sorry, I wasn't working on it lately, but hope I'll get to test some more stuff this weekend.
Will update the status when/if I achieve some progress on the theme

**Comment by Gigas002** [comment]
@kRHYME7 Sorry to disturb you. Just wanted to notify, that theme is in kinda working state, since you've been interested in it too. I've been using it for 2 or 3 months and didn't experienced major issues
https://github.com/Gigas002/hyprdots-sweet

**Reply by kRHYME7** [reply]
We got discord BTW interested in joining?

we got major major changes

https://discord.gg/4HRWS3RH

**Reply by kRHYME7** [reply]
Oh no 
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/0eca2c03-a67d-40da-9d02-e4df5ea91f7d)

Should have informed you that we got a new way of installing themes. 

The themes are now hosted in a different repo (per branch).

You can use this as a template 

https://github.com/prasanthrangan/hyde-themes 
https://github.com/kRHYME7/hyde-gallery


Themes are now very portable

**Reply by Gigas002** [reply]
Hello! Could you please try out the `hyde` branch? I'm not sure about `kvantum.theme` file though, there wasn't any `xml` files in the theme directory so I didn't add it

**Reply by kRHYME7** [reply]
It will not continue to install the theme without the kvantum.theme. 

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/c1b759a6-f5f3-4d1a-9629-bbdc61041fde)

I initially wrote the themepatcher so it will end with lacking configs.  

You check other themes  too for ideas

**Reply by prasanthrangan** [reply]
I just installed it.. but the icon packs seems to be missing some icons

![image](https://github.com/prasanthrangan/hyprdots/assets/106020512/0084a494-0dae-4df9-974b-496456b2a9fe)

also the headers are missing from theme file -
```
❯ head -1 *.theme
==> hypr.theme <==
exec = gsettings set org.gnome.desktop.interface icon-theme 'Sweet-Rainbow'
==> kitty.theme <==
# kitty theme: Sweet
==> rofi.theme <==
* {
==> waybar.theme <==
@define-color bar-bg rgba(0, 0, 0, 0);
```

I'll try to help fix it maybe tomorrow

**Reply by Gigas002** [reply]
Huge thanks for taking a look at this!
Regarding icons, I think the problem is `Sweet-%color%` icons are mostly for dolphin directories, so we need to apply the `candy-icons` first.

UPD
Pushed fix with headers, and attempt to set two icon themes one by one, hope it helps. I don't know how to generate `kvantum.theme` xml though.
Also I have a small problem with *some* icons in tray, e.g. bluetooth and network ones. I don't know which apps are used here, but on KDE session no problems with these two, correct icons are used...

(left one not themed correctly `network`, next one is `bluetooth` and on right themed `telegram` icon)
![image](https://github.com/prasanthrangan/hyprdots/assets/24297712/33e9eb8b-2bdb-429c-8c61-bc2df2dbae96)

**Reply by Gigas002** [reply]
Ah, I see, the kvantum.theme xml was an svg file. I'll push a fix later today, everything else _should_ work fine IMO

**Reply by kRHYME7** [reply]
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/d0c4b0cb-f6d6-4d14-b8fe-319df56df0d1)

I got errors with themepatcher. (aside from the svg, i touch it)

Can you recompress the tarball ?  tar -tf should provide the one I circled. 

Dunno if themepatcher should tolerate a file tree other than ``` Theme/.... ``` e.g ``` ./Theme/..... ``` Will ask @prasanthrangan  about it

Also it's better  to remove other unrelated tarball

Here's the diff
```
diff --git a/Source/arcs/Cursor_Sweet-cursors.tar.xz b/Source/arcs/Cursor_Sweet-cursors.tar.xz
deleted file mode 100644
index d9ce9f3..0000000
Binary files a/Source/arcs/Cursor_Sweet-cursors.tar.xz and /dev/null differ
diff --git a/Source/arcs/Firefox_Sweet.tar.xz b/Source/arcs/Firefox_Sweet.tar.xz
deleted file mode 100644
index 4741006..0000000
Binary files a/Source/arcs/Firefox_Sweet.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_Sweet-Blue-Filled.tar.xz b/Source/arcs/Icon_Sweet-Blue-Filled.tar.xz
deleted file mode 100644
index 938dd48..0000000
Binary files a/Source/arcs/Icon_Sweet-Blue-Filled.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_Sweet-Blue.tar.xz b/Source/arcs/Icon_Sweet-Blue.tar.xz
deleted file mode 100644
index b7d0ae4..0000000
Binary files a/Source/arcs/Icon_Sweet-Blue.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_Sweet-Mars.tar.xz b/Source/arcs/Icon_Sweet-Mars.tar.xz
deleted file mode 100644
index 4b7f3d4..0000000
Binary files a/Source/arcs/Icon_Sweet-Mars.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_Sweet-Purple-Filled.tar.xz b/Source/arcs/Icon_Sweet-Purple-Filled.tar.xz
deleted file mode 100644
index 5bc4356..0000000
Binary files a/Source/arcs/Icon_Sweet-Purple-Filled.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_Sweet-Purple.tar.xz b/Source/arcs/Icon_Sweet-Purple.tar.xz
deleted file mode 100644
index eb5a7c8..0000000
Binary files a/Source/arcs/Icon_Sweet-Purple.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_Sweet-Rainbow.tar.xz b/Source/arcs/Icon_Sweet-Rainbow.tar.xz
deleted file mode 100644
index eb87612..0000000
Binary files a/Source/arcs/Icon_Sweet-Rainbow.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_Sweet-Red-Filled.tar.xz b/Source/arcs/Icon_Sweet-Red-Filled.tar.xz
deleted file mode 100644
index 0629667..0000000
Binary files a/Source/arcs/Icon_Sweet-Red-Filled.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_Sweet-Red.tar.xz b/Source/arcs/Icon_Sweet-Red.tar.xz
deleted file mode 100644
index 35fb430..0000000
Binary files a/Source/arcs/Icon_Sweet-Red.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_Sweet-Teal-Filled.tar.xz b/Source/arcs/Icon_Sweet-Teal-Filled.tar.xz
deleted file mode 100644
index c972e4c..0000000
Binary files a/Source/arcs/Icon_Sweet-Teal-Filled.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_Sweet-Teal.tar.xz b/Source/arcs/Icon_Sweet-Teal.tar.xz
deleted file mode 100644
index c971346..0000000
Binary files a/Source/arcs/Icon_Sweet-Teal.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_Sweet-Yellow-Filled.tar.xz b/Source/arcs/Icon_Sweet-Yellow-Filled.tar.xz
deleted file mode 100644
index 8897467..0000000
Binary files a/Source/arcs/Icon_Sweet-Yellow-Filled.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_Sweet-Yellow.tar.xz b/Source/arcs/Icon_Sweet-Yellow.tar.xz
deleted file mode 100644
index 18a54ab..0000000
Binary files a/Source/arcs/Icon_Sweet-Yellow.tar.xz and /dev/null differ
diff --git a/Source/arcs/Icon_candy-icons.tar.xz b/Source/arcs/Icon_candy-icons.tar.xz
deleted file mode 100644
index 175a5b7..0000000
Binary files a/Source/arcs/Icon_candy-icons.tar.xz and /dev/null differ
diff --git a/Source/arcs/Sddm_Sweet.tar.xz b/Source/arcs/Sddm_Sweet.tar.xz
deleted file mode 100644
index ade1bb7..0000000
Binary files a/Source/arcs/Sddm_Sweet.tar.xz and /dev/null differ


```

**Reply by Gigas002** [reply]
Thanks! Pushed these changes.

**Reply by kRHYME7** [reply]
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/5e10dabb-db38-4be6-abbc-32d184452c86)


Extra arcs could cause errors

**Reply by kRHYME7** [reply]
Will try to revise the themepatcher to tolerate these errors.

**Reply by kRHYME7** [reply]
BTW if you got some time can also add this theme in this repo 

https://github.com/kRHYME7/hyde-gallery/blob/hyde-gallery/hyde-themes.db So that users will have the chance to choose themes at fresh install ? 

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/7cc76833-decb-493f-9b78-17ce1add0ef5)
